### PR TITLE
fix: printing debye-dipole-moment

### PR DESCRIPTION
**Pull request by wltr-tlrc**
### Summary
I found a bug when printing the debye-dipole-moment. This fix has been discussed with @mrossinek.

### CODE REVIEW

File: qiskit_nature/second_q/problems/electronic_structure_result.py [hunk 1]
```diff
@@ -162,7 +162,13 @@ def total_dipole_moment(self) -> Optional[List[float]]:
     def total_dipole_moment_in_debye(self) -> Optional[List[float]]:
         """Returns total dipole of moment in Debye"""
         tdm = self.total_dipole_moment
-        return [dip / DEBYE if dip is not None else None for dip in tdm]
+        if tdm is None:
```
Comment (woodsp-ibm) [inline qiskit_nature/second_q/problems/electronic_structure_result.py:165]: From what I can see the change is to handle the condition where the dipole is not computed at all and ends up as None which would have failed before. When we do compute a dipole, if Z2 symmetry reduction is also being used, one or more of the X,Y and Z components may not commute with the symmetry of the main operator, against whose ground state we would compute the dipole moment, and in this case that aspect is set at None. The list comprehension that was there dealt with any of those individual parts being None, so None was placed into the result. I think the list comprehension above could just be preceded by the overall None check as below. 
```
tdm = self.total_dipole_moment
if tdm is None:
   return None
return [dip / DEBYE if dip is not None else None for dip in tdm]
```

Comment (mrossinek) [reply qiskit_nature/second_q/problems/electronic_structure_result.py:165]: Sure we can do that, too. I will add this as a suggestion to simplify applying the change

File: qiskit_nature/second_q/problems/electronic_structure_result.py [hunk 2]
```diff
@@ -162,7 +162,13 @@ def total_dipole_moment(self) -> Optional[List[float]]:
...
+            return None
+        tdmd = []
+        for dip in tdm:
+            dipmd = dip / DEBYE
+            tdmd.append(dipmd)
+        return tdmd
```
Comment (mrossinek) [inline qiskit_nature/second_q/problems/electronic_structure_result.py:171]:
```suggestion
        return [dip / DEBYE if dip is not None else None for dip in tdm]
```

**Review by mrossinek — APPROVED** [review]
Thank you, @wltr-tlrc, for your first contribution to Qiskit! :partying_face:

**Review by woodsp-ibm — APPROVED** [review]
LGTM. @wltr-tlrc Thanks as well for taking the time to do a contribution for a fix!

**Pull request merged**